### PR TITLE
adapted seg script to new data format

### PR DESCRIPTION
**Pull request by ramirezmichelle**

### CODE REVIEW

File: segmentation/crop_individuals.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                cv2.imwrite(cropped_wing_path, cv2.cvtColor(cropped_wing, cv2.COLOR_RGB2BGR))
+            except FileNotFoundError:
+                errors.append(cropped_wing_path)
+    
+    print('The following images could encountered errors during cropping/resizing:', errors)
+    return
+
+
+if __name__ == "__main__":
+    main()
```
Comment (egrace479) [inline segmentation/crop_individuals.py:126]:
```suggestion
    main()

```

File: README.md [hunk 2]
```diff
@@ -45,4 +45,10 @@ To predict segmentation masks for beetles imaged, run:
 To remove the background of beetle images using their segmentation masks run:
 `python3 remove_background.py --images <path to images> --masks <path to segmentation masks>`
```
Comment (egrace479) [inline README.md:46]:
```suggestion
```
python3 remove_background.py --images <path to images> --masks <path to segmentation masks>
```
```

File: README.md [hunk 3]
```diff
@@ -45,4 +45,10 @@ To predict segmentation masks for beetles imaged, run:
...
 
+To crop out individual beetles from images run:
+`python3 individual_beetles.py --images <path to group_images> --csv <path to metadata/individual_metadata_full.csv>`
```
Comment (egrace479) [inline README.md:49]:
````suggestion
```
python3 individual_beetles.py --images <path to group_images> --csv <path to metadata/individual_metadata_full.csv>
```
````

File: segmentation/individual_beetles.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        try:
+            cv2.imwrite(row['file_name'], cv2.cvtColor(cropped_img, cv2.COLOR_RGB2BGR))
+        except FileNotFoundError:
+            errors.append(row['file_name'])
+    
+    print('The following images could encountered errors during cropping:', errors)
+    return errors
+
+if __name__ == "__main__":
+    main()
```
Comment (egrace479) [inline segmentation/individual_beetles.py:156]:
```suggestion
    main()

```

File: segmentation/utils.py [hunk 5]
```diff
@@ -101,4 +107,15 @@ def load_batch_images(image_filepaths, color_option=0):
...
+def get_sam_model(device):
+    '''Get the SAM VIT l Model'''
+    model_type = "vit_l"
+    sam_checkpoint = "/home/ramirez.528/BeetlePalooza/sam_vit_l_0b3195.pth"
+    if not os.path.exists(sam_checkpoint):
+        model_url = "https://dl.fbaipublicfiles.com/segment_anything/sam_vit_l_0b3195.pth"
+        sam_checkpoint = wget.download(model_url)
+    sam = sam_model_registry[model_type](checkpoint=sam_checkpoint)
+    sam.to(device=device)
+    return sam
```
Comment (egrace479) [inline segmentation/utils.py:121]:
```suggestion
    return sam

```

File: segmentation/utils.py [hunk 6]
```diff
@@ -101,4 +107,15 @@ def load_batch_images(image_filepaths, color_option=0):
...
         return dataset_images
     
-    return dataset_images
\ No newline at end of file
+    return dataset_images
+
+def get_sam_model(device):
+    '''Get the SAM VIT l Model'''
+    model_type = "vit_l"
+    sam_checkpoint = "/home/ramirez.528/BeetlePalooza/sam_vit_l_0b3195.pth"
```
Comment (egrace479) [inline segmentation/utils.py:115]:
```suggestion
    sam_checkpoint = SAM_CHECKPOINT
```

File: segmentation/utils.py [hunk 7]
```diff
@@ -1,7 +1,13 @@
...
 import cv2
-import numpy as np
+import matplotlib.pyplot as plt
+from ultralytics import YOLO
+from segment_anything import sam_model_registry, SamPredictor
+import torch, torchvision
+import argparse
+import wget
+import ast
+import glob, os
```
Comment (egrace479) [inline segmentation/utils.py:10]:
```suggestion
import glob, os

# saved local SAM checkpoint (same as URL)
SAM_CHECKPOINT = "/home/ramirez.528/BeetlePalooza/sam_vit_l_0b3195.pth"
```

**Review by egrace479 — COMMENTED** [review]
Set the SAM checkpoint as a constant variable at the top of the file for easier update.

**Review by egrace479 — APPROVED** [review]
LGTM!
